nodestream/pipeline/transformers/transformer.py

```diff
@@ -129,3 +131,44 @@ def do_work_on_record(self, record: Any) -> Any:
     @abstractmethod
     async def transform_record(self, record: Any) -> Any:
         raise NotImplementedError
+
+
+class SwitchTransformer(Transformer):
+    __slots__ = (
+        "switch_on",
+        "transformers",
+        "default",
+        "normalization",
+    )
+
+    @staticmethod
+    def guarantee_transformer_from_file_data(file_data):
+        return ClassLoader().load_class(**file_data)
+
+    def __init__(
+        self,
+        switch_on: StaticValueOrValueProvider,
+        cases: Dict[str, dict],
+        default: Dict[str, Any] = None,
+        normalization: Dict[str, Any] = None,
+    ):
+        self.switch_on = ValueProvider.guarantee_value_provider(switch_on)
+        self.transformers = {
+            field_value: self.guarantee_transformer_from_file_data(transformer)
+            for field_value, transformer in cases.items()
+        }
+        self.default = (
+            self.guarantee_transformer_from_file_data(default) if default else None
+        )
+        self.normalization = normalization or {}
+
+    async def transform_record(self, record: Any):
+        context = ProviderContext.fresh(record)
+        key = self.switch_on.normalize_single_value(context, self.normalization)
+        transformer = self.transformers.get(key, self.default)
+        # If we don't have a transformer to handle the record we just yield the record (PassTransformer)
+        if transformer is None:
+            yield record
+        else:
+            async for result in transformer.transform_record(record):
+                yield result
```

Comment:
If we create a `PassTransformer`, we can default `self.default` to an instance of pass transformer and remove the branch from the code.